README.md

```diff
@@ -1,2 +1,67 @@
-# thrift-pool
+# Thrift Pool
+
 A module that wraps thrift interfaces in connection pooling logic to make them more resilient.
+
+The [node thrift code](https://www.npmjs.com/package/thrift) exposes a `service` and `types` file.
+The node thrift client also exposes interfaces to create connections to the thrift server.
+There is no way to use the node thrift library to do connection pooling or to create new connections if existing connections fail.
+
+This library takes in a thrift `service` and wraps the methods with connection pooling logic (based on [node-pool](https://github.com/coopernurse/node-pool)).
+
+
+## Installation
+
+```
+npm install node-thrift-pool
+```
+
+## Usage
+```javascript
+var thrift = require('thrift'),
+  Service = require('./gen-nodejs/Service'),
+  Types = require('./gen-nodejs/types'),
+  thriftPool = require('node-thrift-pool');
+
+var thrift_client = thriftPool(thrift, Service, {host: "localhost", port: 9090});
+
+/*
+ * thrift_client is now an initialized thrift client that uses connection pooling behind the scenes
+ * thrift_client.method(function(err, returned_data){console.log(err, returned_data)});
+ */
+
+```
+
+## Supported pooling options
+
+- `host` - **Required** - The host of the thrift server to connect to
+- `port` - **Required** - The port of the thrift server to connect to
+- `max_connections` - Default: `2` - Max number of connections to keep open
```

Comment:
I'll reword to use same description generic pool uses:
`maximum number of resources to create at any given time
                     optional `

I'd rather not go for something higher, given that this number really depends on the service and what it can handle.  I could set it to 1 to stick to what generic pool was using as default and make it clear to users that they need to set it to something they feel comfortable with.